_projects/2024/100508856/100508856.Rmd

```diff
@@ -0,0 +1,333 @@
+
+
+## Graph to replicate
+
+The images I intend to replicate combine line graphs, maps, and bubble charts, each representing different data characteristics. The line graph is used to show the trend of annual working hours over the past 60 years. The maps provide a more macroscopic view of the changes in global working hours. The bubble chart, incorporating population and national GDP, adds complexity to the data and allows for the observation of more information. Finally, an evolutionary path is drawn based on annual working hours and GDP data. The original image depicted the evolutionary paths of all countries over time, dynamically represented across different regional segments. Here is a link to the original graph:
+[here]
+1.(https://ourworldindata.org/grapher/annual-working-hours-per-worker?time=1870..2017)
+2.(https://ourworldindata.org/grapher/annual-working-hours-per-worker?tab=map&time=2005)
+
+
+## Load libraries and read the database
+
+The first step was to load the libraries, in this case I used these: dplyr, ggplots, ggrepel, tidyr, ggokabeito, tmap and paletteer. The second step was to read the database which was on a csv file.
+
+The first image I replicated is a line graph(Figure 1), in which I selected the average annual working hours of over 60 different countries globally from 1950 to 2017. The original was a dynamic trend graph that changed with the slide of time, and I created the final overall line trend graph. However, due to the excessive number of countries, it appeared quite cluttered at first glance. Therefore, in the second image(Figure 2), I chose a few representative countries to make the chart clearer, whereas the original graph was actually an overlay of different countries' and regions' trend graphs. Subsequently, there are four map images(Figure 3-6). Each map depicts the state of annual working hours in different regions globally during various years of this period, using shades of color to represent the length of working hours
+
+
+## First two graphs
+
+The code for this visual representation can be divided into four parts:
+
+1. Modify column names after reading the corresponding data table
+2. Set the X-axis as the year, the Y-axis as the working time, and the colors are differentiated by region
+3. Modify the title to "Annual Working Hours per Country from 1950 to 2017"
+4. Modify the x-axis of the plot
+
+
+```{r}
+
+
+# Figure 1
+library(ggplot2)
+library(dplyr)
+library(ggrepel)
+library(tidyr)
+devtools::install_github("malcolmbarrett/ggokabeito")
+library(ggokabeito)
+library(paletteer)
+working_hours<-read.csv("C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/annual working hours/annual-working-hours-per-worker.csv")
```

Comment:
The CSV is missing in the pull request. Please add this file, and then these lines should be just `read.csv("annual-working-hours-per-worker.csv")`.